src/pages/academy/notiPreference/NotiPreference.tsx

```diff
@@ -0,0 +1,184 @@
+import { Button, H1, Intent } from '@blueprintjs/core';
+import { GridApi, GridReadyEvent, ValueFormatterFunc } from 'ag-grid-community';
+import { AgGridReact } from 'ag-grid-react';
+import { cloneDeep } from 'lodash';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import {
+  fetchConfigurableNotificationConfigs,
+  updateNotificationPreferences
+} from 'src/commons/application/actions/SessionActions';
+import {
+  NotificationConfiguration,
+  NotificationPreference,
+  TimeOption
+} from 'src/commons/application/types/SessionTypes';
+import ContentDisplay from 'src/commons/ContentDisplay';
+import { useTypedSelector } from 'src/commons/utils/Hooks';
+
+import BooleanCell from './subcomponents/BooleanCell';
+import SelectCell from './subcomponents/SelectCell';
+
+const NotiPreference: React.FC = () => {
+  const gridApi = React.useRef<GridApi>();
+
+  const dispatch = useDispatch();
+  const session = useTypedSelector(state => state.session);
+
+  const [hasChanges, setHasChanges] = useState<boolean>(false);
+
+  const configurableNotificationConfigs = React.useRef(
+    session.configurableNotificationConfigs
+  ) as React.MutableRefObject<NotificationConfiguration[]>;
```

Comment:
The UI bug is caused by this typecast. This should not be the correct way to do things. I will fix this, no worries.

Reply:
Thanks!